crates/turbopack-ecmascript/src/references/esm/meta.rs

```diff
@@ -0,0 +1,75 @@
+use anyhow::Result;
+use swc_core::{
+    common::DUMMY_SP,
+    ecma::ast::{Expr, Ident},
+    quote,
+};
+use turbo_tasks_fs::FileSystemPathVc;
+use turbopack_core::chunk::ChunkingContextVc;
+
+use crate::{
+    code_gen::{CodeGenerateable, CodeGenerateableVc, CodeGeneration, CodeGenerationVc},
+    create_visitor, magic_identifier,
+    references::{as_abs_path, esm::base::insert_hoisted_stmt, AstPathVc},
+};
+
+#[turbo_tasks::value(shared)]
+#[derive(Hash, Debug)]
+pub struct ImportMetaRef {
+    path: FileSystemPathVc,
+    initialize: bool,
```

Comment:
I think this `initialize` flag is a bit weird.

We could split the logic into `ImportMeta` and `ImportMetaBinding`, similar to `EsmAssetReference` and `EsmBinding`.

`ImportMeta` would only inject the declaration.
`ImportMetaBinding` would only replace the expression.

But if that is too much work, I will also accept the initialize flag...